scripts/fetch_data.py

```diff
@@ -0,0 +1,274 @@
+                                             model=model,
+                                             emission=args['emission'],
+                                             time=args['time'],
+                                             area=args['area'],
+                                             variable=args['variable']['max'],
+                                             timescale=args['timescale'],
+                                             cell_method=args['cell_method']))
+        try:
+            return mean([tasmin, tasmax])
+        except TypeError as e:
+            logger.debug(('Unable to get mean of tasmin: {} and tasmax: {} in '
+                          'model: {} error: {}')
+                         .format(tasmin, tasmax, model, e))
+            return None
+
+    elif args['variable'] == 'fdETCCDI':
+        fd = filter_by_period(args['spatial'], args['dates'],
+                              multistats(sesh,
+                                         ensemble_name='p2a_files',
+                                         model=model,
+                                         emission=args['emission'],
+                                         time=args['time'],
+                                         area=args['area'],
+                                         variable=args['variable'],
+                                         timescale=args['timescale'],
+                                         cell_method=args['cell_method']))
+        try:
+            return get_nffd(fd, args['time'], args['timescale'])
+        except TypeError as e:
+            logger.warning('Unable to compute nffd from fd with {} error: {}'
+                           .format(fd, e))
+            return None
+
+    else:
+        return filter_by_period(args['spatial'], args['dates'],
+                                multistats(sesh,
+                                           ensemble_name='p2a_files',
+                                           model=model,
+                                           emission=args['emission'],
+                                           time=args['time'],
+                                           area=args['area'],
+                                           variable=args['variable'],
+                                           timescale=args['timescale'],
+                                           cell_method=args['cell_method']))
+
+
+def get_models(sesh, percentile):
+    """Return a list of models needed to compute the percentile"""
+    historical_baseline = 'anusplin'
+    if percentile == 'hist':
+        return [historical_baseline]
+    else:
+        # return all models EXCEPT for the historical baseline
+        all_models = models(sesh, ensemble_name='p2a_files')
+        all_models.remove(historical_baseline)
+        return all_models
+
+
+def prep_args(variable, time_of_year, temporal, spatial, percentile, area,
```

Comment:
I'd lean towards decomposing this into several functions that handle their own area of expertise. E.g.

```
def translate_variable(varname):
    pass


def translate_time_of_year(time_of_year):
    pass
```

Etc. Since, really, none of the parameters have any bearing on the others.

Reply:
Good idea, my goal was to try and reduce the massive number of parameters that would get passed from function to function.  However, I agree that common items could be factored out.